CMakeLists.txt

```
@@ -50,6 +50,25 @@ endif()
 
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+execute_process(
+    COMMAND git describe --dirty --always 2>/dev/null
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    OUTPUT_VARIABLE OT_COMM_GIT_REVISION OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+if(OT_COMM_GIT_REVISION)
+    set(OT_COMM_VERSION "${PROJECT_VERSION}+${OT_COMM_GIT_REVISION}")
+else()
+    set(OT_COMM_VERSION "${PROJECT_VERSION}")
+endif()
```

Allow customize version.
```suggestion
if(NOT OT_COMM_VERSION)
    if(OT_COMM_GIT_REVISION)
        set(OT_COMM_VERSION "${PROJECT_VERSION}+${OT_COMM_GIT_REVISION}")
    else()
        set(OT_COMM_VERSION "${PROJECT_VERSION}")
    endif()
endif()
```

---

Are there cases that we want need to specify a version other that the project version? I think there is no other source of the version.